gattc_darwin.go

```diff
@@ -0,0 +1,121 @@
+package bluetooth
+
+import (
+	"errors"
+	"time"
+
+	"github.com/JuulLabs-OSS/cbgo"
+)
+
+// DiscoverServices starts a service discovery procedure. Pass a list of service
+// UUIDs you are interested in to this function. Either a slice of all services
+// is returned (of the same length as the requested UUIDs and in the same
+// order), or if some services could not be discovered an error is returned.
+func (d *Device) DiscoverServices(uuids []UUID) ([]DeviceService, error) {
+	cbuuids := []cbgo.UUID{}
+	for _, u := range uuids {
+		uuid, _ := cbgo.ParseUUID(u.String())
+		cbuuids = append(cbuuids, uuid)
+	}
+
+	d.prph.DiscoverServices(cbuuids)
+
+	// wait on channel for service discovery
+	select {
+	case <-d.servicesChan:
+		svcs := []DeviceService{}
+		for _, dsvc := range d.prph.Services() {
+			uuid, _ := ParseUUID(dsvc.UUID().String())
+			svc := DeviceService{
+				UUID:    uuid,
+				device:  d,
+				service: dsvc,
+			}
+			svcs = append(svcs, svc)
+		}
+		return svcs, nil
+	case <-time.NewTimer(10 * time.Second).C:
+		return nil, errors.New("timeout on DiscoverServices")
```

Comment:
I suspect this doesn't really time out either, it either succeeds or the entire connection is broken.

Reply:
Same as above comment.